fenix/app/src/main/java/org/mozilla/fenix/compose/SwitchWithLabel.kt

```diff
@@ -0,0 +1,179 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.SwitchDefaults
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.compositeOver
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.theme.FirefoxTheme
+import androidx.compose.material.Switch as MaterialSwitch
+
+private const val DISABLED_ALPHA = 0.5f
+
+/**
+ * UI for a switch with label that can be on or off.
+ *
+ * @param checked Whether or not the switch is checked.
+ * @param onCheckedChange Invoked when Switch is being clicked, therefore the change of checked
+ * state is requested.
+ * @param label Text to be displayed next to the switch.
+ * @param modifier Modifier to be applied to the switch layout.
+ * @param enabled Whether the switch is enabled or grayed out.
+ */
+@Composable
+fun SwitchWithLabel(
+    checked: Boolean,
+    onCheckedChange: ((Boolean) -> Unit),
+    label: String,
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+) {
+    Row(
+        modifier = modifier,
+        horizontalArrangement = Arrangement.spacedBy(16.dp),
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        Text(
+            text = label,
+            color = if (enabled) {
+                FirefoxTheme.colors.textPrimary
+            } else {
+                FirefoxTheme.colors.textDisabled
+            },
+            style = FirefoxTheme.typography.subtitle1,
+            modifier = Modifier.weight(1f),
+        )
+
+        Switch(
+            checked = checked,
+            onCheckedChange = onCheckedChange,
+            enabled = enabled,
+        )
+    }
+}
+
+/**
+ * UI for a switch that can be on or off.
+ *
+ * @param checked Whether or not the switch is checked.
+ * @param onCheckedChange Invoked when Switch is being clicked, therefore the change of checked
+ * state is requested.
+ * @param modifier Modifier to be applied to the switch layout.
+ * @param enabled Whether the switch is enabled or grayed out.
+ */
+@Composable
+private fun Switch(
+    checked: Boolean,
+    onCheckedChange: ((Boolean) -> Unit),
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+) {
+    MaterialSwitch(
+        checked = checked,
+        onCheckedChange = onCheckedChange,
+        modifier = modifier,
+        enabled = enabled,
+        colors = SwitchDefaults.colors(
+            uncheckedThumbColor = FirefoxTheme.colors.formOff,
+            uncheckedTrackColor = FirefoxTheme.colors.formSurface,
+            checkedThumbColor = FirefoxTheme.colors.formOn,
+            checkedTrackColor = FirefoxTheme.colors.formSurface,
+            disabledUncheckedThumbColor = FirefoxTheme.colors.formOff
+                .copy(alpha = DISABLED_ALPHA)
+                .compositeOver(FirefoxTheme.colors.formSurface),
```

Comment:
where is the alpha value derived from? 

We may not need to do this copy copying + alpha logic, as`MaterialSwitch` is already applying an alpha via `uncheckedTrackAlpha` and `checkedTrackAlpha` as part of the `SwitchDefaults.colors` API

Reply:
The alpha is from the [designs](https://www.figma.com/file/ujdiReypSypu2GhRgzArea/Android-Components?type=design&node-id=3031-17475&mode=design&t=liRd1rKuUxEhzCcZ-0). 

It seems we do, otherwise it composites over MaterialTheme.colors.surface, which results in a different color from the spec. 

Tangential, An idea would be to map the theme colors to material as well so when we use some material component like dialog/calendar it automatically uses our surface/primary colors, the basic ones which are easily mappable.